go/vt/vtorc/inst/shard_dao.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inst
+
+import (
+	"errors"
+
+	"vitess.io/vitess/go/vt/external/golib/sqlutils"
+	"vitess.io/vitess/go/vt/logutil"
+	"vitess.io/vitess/go/vt/topo"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+	"vitess.io/vitess/go/vt/vtorc/db"
+)
+
+// ErrShardNotFound is a fixed error message used when a shard is not found in the database.
+var ErrShardNotFound = errors.New("shard not found")
+
+// ReadShardPrimaryInformation reads the vitess shard record and gets the shard primary alias and timestamp.
+func ReadShardPrimaryInformation(keyspaceName, shardName string) (primaryAlias string, primaryTimestamp string, err error) {
+	if err = topo.ValidateKeyspaceName(keyspaceName); err != nil {
+		return
+	}
+	if _, _, err = topo.ValidateShardName(shardName); err != nil {
+		return
+	}
+
+	query := `
+		select
+			primary_alias, primary_timestamp
+		from
+			vitess_shard
+		where keyspace = ? and shard=?
```

Comment:
Just for the sake of consistency
```suggestion
		where keyspace=? and shard=?
```

Reply:
Done.